components/backends/trtllm/performance_sweeps/scripts/bench.sh

```diff
@@ -96,48 +96,39 @@ if [ -f "${artifacts_dir}/deployment_config.json" ]; then
 fi
 echo "${deployment_config}" > "${artifacts_dir}/deployment_config.json"
 
-# TODO: This is a temporary fix to check if the server is up.
-# We should use a more robust health check mechanism.
-
-# Loop up to 50 times
+# Wait for server to become healthy (up to 50 attempts)
+failed=true
 for ((i=1; i<=50; i++)); do
-    # Run curl and capture response and HTTP code
-    response=$(curl -s -w "\n%{http_code}" "${hostname}:${port}/v1/chat/completions" \
-      -H "Content-Type: application/json" \
-      -d "{
-        \"model\": \"${model}\",
-        \"messages\": [
-           {
-            \"role\": \"user\",
-            \"content\": \"Tell me a story as if we were playing dungeons and dragons.\"
-           }
-        ],
-        \"stream\": true,
-        \"max_tokens\": 30
-      }")
-
-    # Extract HTTP code
+    response=$(curl -s -w "\n%{http_code}" "${hostname}:${port}/health")
     http_code=$(echo "$response" | tail -n1)
-
-    if [ "$http_code" = "200" ]; then
-        echo "Success on attempt $i"
-        # Optional: Print the response body (excluding HTTP code)
-        echo "$response" | sed '$d'
-        break
-    else
-        echo "Attempt $i failed (HTTP $http_code)."
-
-        # Wait: 100 seconds after first failure, 10 seconds after subsequent
-        if [ "$i" -eq 1 ]; then
-            sleep 300
+    body=$(echo "$response" | sed '$d')
+
+    if [[ "$http_code" == "200" ]] && echo "$body" | grep -q '"status":"healthy"' && echo "$body" | grep -q '"endpoints":\[[^]]*"dyn://dynamo.tensorrt_llm.generate"'; then
+        if [[ "$kind" == *disagg* ]]; then
+            etcd_response=$(curl -s -L http://localhost:2379/v3/kv/range -X POST -d '{"key": "Lw==", "range_end": "AA=="}')
+            etcd_keys=$(echo "$etcd_response" | jq -r '.kvs // [] | .[] | .key |= @base64d | select(.key | contains("tensorrt_llm_next")) | .key')
```

Comment:
I believe health check output will also contain tensorrt_llm_next btw, I don't think you need a separate etcd check and can use /health for both